docs/cyber.md

```diff
@@ -0,0 +1,80 @@
+# access anomalies: [complement_access.py](../src/main/python/mmlspark/cyber/ml/access_anomalies/complement_access.py)
+- [Talk at European Spark Conference 2019](https://databricks.com/session_eu19/cybermltoolkit-anomaly-detection-as-a-scalable-generic-service-over-apache-spark)
+- [(Internal Microsoft) Talk at MLADS November 2018](https://resnet.microsoft.com/video/42395)
+- [(Internal Microsoft) Talk at MLADS June 2019](https://resnet.microsoft.com/video/43618)
+
+1. [ComplementAccessTransformer](../src/main/python/mmlspark/cyber/ml/access_anomalies/complement_access.py)
+   is a SparkML [Transformer](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Transformer.html).
+   Given a dataframe it returns a new dataframe containing new access patterns sampled from
+   the set of possible access patterns which did not occur in the given dataframe
+   (i.e., it returns a sample from the complement set).
+
+# feature engineering: [indexers.py](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+1. [IdIndexer](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Estimator](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Estimator.html).
+   Given a dataframe, it creates an IdIndexerModel (described next) for categorical features which
+   contains the information to map each partition and column seen in the given dataframe to an id.
+   for each partition or one consecutive range for all partition and column values.
+2. [IdIndexerModel](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Transformer](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Transformer.html).
+   Given a dataframe maps each partition and column field to a consecutive integer id.
+   Partitions or column values not encountered in the estimator are mapped to 0.
+   The model can operate in two modes, either create consecutive integer id independently 
+3. [MultiIndexer](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
```

Comment:
Please consider using the `MultiColumnAdapter` already in the library here:
https://github.com/Azure/mmlspark/blob/master/src/main/scala/com/microsoft/ml/spark/stages/MultiColumnAdapter.scala